Cubical/Algebra/NatSolver/EvaluationHomomorphism.agda

```diff
@@ -1,91 +1,66 @@
 {-# OPTIONS --safe #-}
-module Cubical.Algebra.RingSolver.EvaluationHomomorphism where
+module Cubical.Algebra.NatSolver.EvaluationHomomorphism where
```

Comment:
The corresponding module in the ring solver is called `EvalHom`. Can we use the same name for both? (I don't care which one.)